### PR TITLE
feat(ops): Implement Context-Aware Histidine Protonation and Refine pH Handling

### LINKED ISSUE #37: Implement Context-Aware Histidine Protonation via Salt Bridge Detection
### Description:

Histidine's protonation state is highly sensitive to its local ionic environment. To improve chemical fidelity, we need to detect salt bridges between histidine and nearby carboxylate groups (ASP, GLU, C-termini) and automatically assign the doubly-protonated HIP state when appropriate. This context-aware logic should operate within the pH 6.0–8.5 range and integrate seamlessly across CLI, WASM, and Web interfaces.

### Tasks:

- [x] **Core Library (`crates/bio-forge/src/ops/hydro.rs`)**:
  - [x] Add `HIS_SALT_BRIDGE_PH_MAX` (8.5) and `SALT_BRIDGE_THRESHOLD` (4.0 Å) constants.
  - [x] Extend `HydroConfig` with `detect_his_salt_bridges: bool` field (default `true`).
  - [x] Implement `build_carboxylate_grid()` to index ASP/GLU/C-terminal oxygens.
  - [x] Implement `his_forms_salt_bridge()` to detect ND1/NE2 proximity to carboxylates.
  - [x] Update `determine_protonation_state()` logic for pH-dependent salt bridge override.
  - [x] Add unit tests covering edge cases (pH bounds, ASH vs ASP, distance cutoffs).

- [x] **CLI (`crates/bio-forge/src/bin/commands/hydro.rs`)**:
  - [x] Add `--no-salt-bridges` flag to `HydroArgs`.
  - [x] Update `run()` to pass flag into `HydroConfig`.

- [x] **WASM Bindings (`crates/bio-forge-wasm/src/lib.rs`)**:
  - [x] Add `detectHisSaltBridges: bool` to WASM `HydroConfig` struct.
  - [x] Update `Default` impl and `From<HydroConfig>` conversion.
  - [x] Verify generated TypeScript types in `.d.ts`.

- [x] **Web Application (`web/src/`)**:
  - [x] Update `HydroSettings` interface in `core/pipeline/config.ts`.
  - [x] Add field to `DEFAULT_HYDRO_SETTINGS` and `toHydroConfig()`.
  - [x] Add Checkbox control in `features/pipeline/step-hydro.tsx`.

- [x] **Documentation**:
  - [x] Update `ARCHITECTURE.md` Hydro Pipeline flowchart and description.
  - [x] Add `--no-salt-bridges` flag documentation to `MANUAL.md`.
  - [x] Update `HydroConfig` table in `API.md` with new field.

### PR DESCRIPTION
**Pull request by TKanX**
### Summary:

Significantly enhances the hydrogenation pipeline (`ops::hydro`) by introducing context-aware histidine protonation logic. It now detects potential salt bridges between histidine residues and nearby carboxylate groups (ASP⁻, GLU⁻, C-terminal COO⁻), automatically converting them to the doubly protonated HIP state when geometrically favorable. Additionally, the pH handling logic has been refined to allow users to opt-out of automatic protonation state changes by omitting the `--ph` flag, preserving original residue names while still adding hydrogens.

### Changes:

- **Enhanced Histidine Protonation:**
  - Implemented `his_forms_salt_bridge` to detect geometric proximity (default 4.0Å) between HIS nitrogens (ND1/NE2) and carboxylate oxygens.
  - Added logic to automatically assign the HIP state when a salt bridge is detected, overriding pH or strategy settings (unless disabled).
  - Introduced `his_salt_bridge_protonation` flag to `HydroConfig` to control this behavior.

- **Refined pH Handling Logic:**
  - Decoupled `add_hydrogens` from mandatory pH inputs.
  - If `target_ph` is `None`:
    - Non-histidine residues retain their original names (e.g., ASP stays ASP).
    - Histidines are only modified if salt bridge detection triggers HIP assignment.
    - Termini assume default physiological states (protonated N-term, deprotonated C-term) unless overridden by explicit pKa checks.
  - Introduced explicit pKa constants for all titratable residues (ASP, GLU, LYS, ARG, CYS, TYR, HIS).

- **Optimized Grid Structures:**
  - Created specialized grid builders: `build_carboxylate_grid` for salt bridge detection (tracking COO⁻ groups) and `build_acceptor_grid` for H-bond network analysis.
  - These grids are computed in parallel using `rayon` when available.

- **CLI and WASM Updates:**
  - Updated the `hydro` command to accept `--no-his-salt-bridge`.
  - Updated WASM bindings to expose the new configuration option.
  - Enhanced UI components (`Input`, `Select`, `Checkbox`) to support description text for better user guidance on complex options.
  - Updated `MANUAL.md` and `ARCHITECTURE.md` to reflect the new multi-phase protonation logic.

### CODE REVIEW

File: crates/bio-forge/src/ops/hydro.rs [hunk 1]
```diff
@@ -134,203 +176,225 @@ pub fn add_hydrogens(structure: &mut Structure, config: &HydroConfig) -> Result<
...
-            config.his_strategy,
-            grid,
-            indices,
-        ));
+        return Some("HIP".to_string());
     }
 
-    None
+    config.target_ph?;
+
```
[the remark this answers is not in the document]
Comment (TKanX) [reply crates/bio-forge/src/ops/hydro.rs:327]: No, It will not.

`crates/bio-forge/src/io/context.rs` => `register_alias!("HIS", "HID");`
